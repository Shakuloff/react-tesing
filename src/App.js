// import logo from './logo.svg';
// import './App.css';
// import { useEffect, useState } from 'react';

// function App() {
//   const [data, setData] = useState(null)
//   const [toggle, setToggle] = useState(false)
//   const [value, setValue] = useState('')

//   const onClicked = () => setToggle(prev => !prev)

//   useEffect(() => {
//     setTimeout(() => {
//       setData({})
//     }, 100)
//   }, [])

//   return (
//     <div>
//       <h1 data-testid="value-elem">{value}</h1>
//       {toggle === true && <div data-testid="toggle-elem">Переключён</div>}
//       {data && <div style={{ color: 'red' }}>Данные</div>}
//       <h1>Привет мир!</h1>
//       <button data-testid="toggle-btn" onClick={onClicked}>Кликни меня</button>
//       <input onChange={e => setValue(e.target.value)} type='text' placeholder='значение.....' />
//     </div >
//   );
// }

// export default App;

import React from 'react'
import { BrowserRouter, Link, Route, Routes } from 'react-router-dom'
import MainPage from './pages/MainPage'
import AboutPage from './pages/AboutPage'

const App = () => {
  return (
    <BrowserRouter>
      <Link data-testid="main-link" to={'/'}>main</Link>
      <Link data-testid="about-link" to={'/about'}>about</Link>
      <Routes>
        <Route path='/' element={<MainPage />}></Route>
        <Route path='/about' element={<AboutPage />}></Route>
      </Routes>
    </BrowserRouter>
  )
}

export default App
