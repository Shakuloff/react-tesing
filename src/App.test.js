import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

// describe("Первый тест", () => {
//   test('common', () => {
//     render(<App />);
//     const helloWorldElem = screen.getByText(/Привет мир!/i)
//     const btn = screen.getByRole("button")
//     const input = screen.getByPlaceholderText(/значение.../i)
//     expect(helloWorldElem).toBeInTheDocument()
//     expect(btn).toBeInTheDocument()
//     expect(input).toBeInTheDocument()
//     expect(input).toMatchSnapshot()
//   });
//   test('query', () => {
//     render(<App />);
//     const helloWorldElem = screen.queryByText(/Привет мир2!/i)
//     expect(helloWorldElem).toBeNull()
//   });
//   test('find', async () => {
//     render(<App />);
//     const helloWorldElem = await screen.findByText(/данные/i)
//     expect(helloWorldElem).toBeInTheDocument()
//     expect(helloWorldElem).toHaveStyle({ color: "red" })
//   });
// })

describe.skip("Клик", () => {
  test("переключение", () => {
    render(<App />);
    const btn = screen.getByTestId('toggle-btn')
    // const toggleDiv = screen.queryByTestId("toggle-elem")
    expect(screen.queryByTestId('toggle-elem')).toBeNull()
    fireEvent.click(btn)
    expect(screen.getByTestId('toggle-elem')).toBeInTheDocument()
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeNull()
  })

  test("поле ввода", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/значение.../i)
    expect(screen.queryByTestId('value-elem')).toContainHTML('')
    // ! искуственно
    fireEvent.input(input, {
      target: { value: 'красный' }
    })
    // ! приближённо к действиям пользователя
    userEvent.type(input, "красный")
    expect(screen.getByTestId('value-elem')).toContainHTML('красный')
  })
})


