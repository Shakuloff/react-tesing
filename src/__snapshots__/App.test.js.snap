// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  placeholder="значение....."
  type="text"
/>
`;

exports[`Первый тест common 1`] = `
<input
  placeholder="значение....."
  type="text"
/>
`;

exports[`Первый тест renders learn react link 1`] = `
<input
  placeholder="значение....."
  type="text"
/>
`;
